pkg/api/publishWorker.go

```diff
@@ -134,7 +136,9 @@ func (p *PublishWorker) publishStagedEnvelope(stagedEnv queries.StagedOriginator
 
 	// Try to delete the row regardless of if the gateway envelope was inserted elsewhere
 	deleted, err := q.DeleteStagedOriginatorEnvelope(context.Background(), stagedEnv.ID)
```

Comment:
We can also pass in `p.ctx` here so that this will fail if the server is in the process of shutting down

Reply:
Totally missed this one, good catch